src/tests/build.proj

```diff
@@ -347,131 +334,71 @@
-        Inputs="@(_AotInputAssemblies)">
-        <Output
-            TaskParameter="Filtered"
-            ItemName="_AotInputAssembliesDedup"/>
-    </RemoveDuplicates>
-    <Message Text="RunAOTCompilation $(RunAOTCompilation)" Importance="High"/>
-    <MonoAOTCompiler Condition="'$(RunAOTCompilation)' == 'true'"
-        CompilerBinaryPath="@(MonoAotCrossCompiler->WithMetadataValue('RuntimeIdentifier','$(TargetOS.ToLowerInvariant())-$(TargetArchitecture.ToLowerInvariant())'))"
-        OutputDir="$(_MobileIntermediateOutputPath)"
-        Mode="$(_AOTMode)"
-        OutputType="AsmOnly"
-        Assemblies="@(_AotInputAssembliesDedup)"
-        AotModulesTablePath="$(BundleDir)\modules.m"
-        AotModulesTableLanguage="ObjC"
-        IntermediateOutputPath="$(_MobileIntermediateOutputPath)"
-        AdditionalAssemblySearchPaths="$(CORE_ROOT)">
-        <Output TaskParameter="CompiledAssemblies" ItemName="_AppleAssembliesInternal" />
-    </MonoAOTCompiler>
-
-    <ItemGroup>
-      <_AppleAssembliesInternal Include="@(_AOT_InternalForceInterpretAssemblies)" />
-    </ItemGroup>
-
-    <RemoveDuplicates
-        Inputs="@(_AppleAssembliesInternal)">
-        <Output
-            TaskParameter="Filtered"
-            ItemName="_AppleAssembliesInternalDedup"/>
-    </RemoveDuplicates>
 
-    <AppleAppBuilderTask
-        TargetOS="$(TargetOS)"
-        ProjectName="$(AssemblyName)"
-        AppDir="$(BuildDir)"
-        MonoRuntimeHeaders="$(MicrosoftNetCoreAppRuntimePackNativeDir)/include/mono-2.0"
-        Assemblies="@(_AppleAssembliesInternalDedup)"
-        Arch="$(TargetArchitecture)"
-        OutputDirectory="$(BundleDir)"
-        Optimized="True"
-        DevTeamProvisioning="$(DevTeamProvisioning)"
-        BuildAppBundle="True"
-        GenerateXcodeProject="True"
-        UseConsoleUITemplate="True"
-        ForceAOT="$(RunAOTCompilation)"
-        RuntimeComponents="$(RuntimeComponents)"
-        DiagnosticPorts="$(DiagnosticPorts)"
-        InvariantGlobalization="true"
-        ForceInterpreter="$(MonoForceInterpreter)"
-        EnableAppSandbox="$(EnableAppSandbox)">
-      <Output TaskParameter="AppBundlePath" PropertyName="AppBundlePath" />
-      <Output TaskParameter="XcodeProjectPath" PropertyName="XcodeProjectPath" />
-    </AppleAppBuilderTask>
-
-    <!-- Apparently MSBuild cannot move directories and recursively copying a
-         a directory requires writing some sort of recursive traversal
-         logic yourself. -->
-    <ItemGroup>
-      <RecursiveCopyHack Include="$(AppBundlePath)/**/*.*" />
-    </ItemGroup>
-    <MakeDir Directories="$(FinalPath)" />
-    <Copy SourceFiles="@(RecursiveCopyHack)" DestinationFolder="$(FinalPath)/%(RecursiveDir)" />
-    <RemoveDir Directories="$(AppBundlePath)" />
-    <Message Importance="High" Text="App: $(FinalPath)" />
+    <CustomTask EntryPoints="@(EntryPoints)" RunScriptCommand="$(RunScriptCommand)" AppName="$(Category)" Condition="'$(BuildTestsOnHelix)' == 'true'">
```

Comment:
nit: I don't have a good suggestion, but I think the task should have a more meaningful name.
Maybe something like `UpdateRunScriptToHandleMultipleEntryPoints`

Reply:
Of course, I missed that.